source/develop/release-management/features/ovirt-ovn-provider.md

```diff
@@ -0,0 +1,334 @@
+*   POST Subnet - not implemented (no OVN implementation yet)
+*   DELETE Subnet - not implemented (no OVN implementation yet)
+
+The oVirt OVN provider can be located anywhere in the system. The address of the
+provider is specified when the provider is added to oVirt, so the two can be on
+any two hosts if needed.
+The provider contacts OVN using a web socket, so the two can also be located on
+separate hosts.
+
+### oVirt OVN VIF Driver
+
+The oVirt OVN VIF driver connects the oVirt VNIC to the proper OVS bridge 
+and OVN logical network. When an OVN port (added by the provider) is plugged
+into an OVS bridge, OVN Controller will retrieve information about the port
+and the logical network it belongs to from OVN south DB, and create the required
+open flows on the local OVS switch. The OVN controller will also update the
+binding tables on the OVN south DB, causing all other OVN controllers on other
+hosts to update their open flow tables, to allow connections to the newly added
+port (if ports belonging to this logical network are present on this OVS switch).
+The oVirt OVN VIF driver will also take care of providing the data about the
+OVN ports needed by the GetCapabilites and GetStatistics VDSM commands, and
+protect the OVN switches and ports during the SetupNetworks command.
+The oVirt OVN VIF driver must be located on each oVirt host which supports
+OVN. The VIF driver is invoked by VDSM hooks, contacting the OVN databases to
+fetch the required data and updating the hook XML passed libvirt or the
+oVirt engine.
+
+The following vdsm hooks will need to be implemented:
+
+*   before_device_create
+*   before_nic_hotplug
+*   after_vm_start
+*   after_get_caps
+*   after_get_stats
+
+## OVN provider lifecycle
+
+### Adding Network
+
+The ovn representation of an ovirt network is a logical switch.
+To add an oVirt network, a logical switch must be added to the OVN north db.
+This is done by adding a record to the Logical_Switch table (equivalent of
+command: ovn-nbctl ls-add <network name>).
+
+### Removing Network
+
+To remove an oVirt network, the equivalent logical switch record must be removed from the OVN north db.
+This is done by removing a record from the Logical_Switch table
+(equivalent of command: ovn-nbct ls-del <network-name>)
+
+All logical ports belonging to this logical switch must be removed from the logical switch prior to this.
+
+### Vnic add
+
+The OVN representation of an oVirt VNIC is a logical port. 
+When a nic is added in ovirt (just added, not plugged in), a logical port must be added to the north db
+(we will actually use lazy initialization and create the port just before the port is plugged in).
+
+The is added by adding a row to the Logical_Switch_Port table in the OVN north db and
+associate it with the mac address of the oVirt nic by setting the appropriate value in
+the record.
+Equivalent of commands:
+ovn-nbctl lsp-add <lswtich name> <lport name (vif id)>
```

Comment:
Please use rich text capabilities. I think that quoting command line with "``" would make the page look cooler.